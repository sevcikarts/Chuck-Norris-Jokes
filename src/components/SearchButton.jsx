import React from "react";
import { useDispatch } from "react-redux";
import Button from "@material-ui/core/Button";
import { fetchData } from ".././redux/actions/dataActions";

const SearchButton = ({ setQuery, setError }) => {
  const dispatch = useDispatch();

  return (
    <div className="inControll">
      <Button
        className="button"
        size="large"
        variant="contained"
        color="primary"
        style={{ width: "300px", fontSize: "20px" }}
        onClick={() => dispatch(fetchData(), setQuery(""), setError(""))}
      >
        Random Joke
      </Button>
    </div>
  );
};

export default SearchButton;
