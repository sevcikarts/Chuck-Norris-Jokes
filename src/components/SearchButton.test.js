import React from "react";
import Enzyme, { mount, render, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import SearchButton from "./SearchButton.jsx";
import configureStore from "redux-mock-store";
import renderer from 'react-test-renderer';


Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore([]);


describe('My Connected React-Redux Component', () => {
  let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });
 
    component = renderer.create(
      <Provider store={store}>
        <SearchButton />
      </Provider>
    );
  });
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });



});