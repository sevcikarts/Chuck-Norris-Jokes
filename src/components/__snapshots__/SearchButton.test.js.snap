// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="inControll"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained button MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "fontSize": "20px",
        "width": "300px",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Random Joke
    </span>
  </button>
</div>
`;
